### PR TITLE
Test coverage increase

### PR DESCRIPTION
**Pull request by beardypig**
A current work in progress, increasing the test coverage. 

Base increase ~10% with 25dbc76.

- [ ] 1 %
- [ ] 2 %
- [ ] 5 %
- [ ] 10 %

### CODE REVIEW

**Comment by gravyboat** [comment]
Nice @beardypig. I was actually thinking about this a few days ago when reviewing your mpeg dash changes!

**Comment by beardypig** [comment]
Again this one -11%, doesn't make sense to me ... looking in to it :/

**Comment by back-to** [comment]
for the **-11%** error

---

in **.coveragerc**, if you change

```
[run]
source =
    streamlink
    streamlink_cli
```

to

```
[run]
source =
    src
```

**master** will change from `33%` to `47%`

**test-coverage** will be 48%

---

```
pytest --cov=src/
```

this is the same but the other might be cleaner

---

all the `0%` plugins will change because `can_handle_url` already run at some point,

but I am not sure how accurate the results are.

File: tests/__init__.py [hunk 1]
```diff
@@ -0,0 +1,8 @@
+import sys
+
+if sys.version_info[0:2] == (2, 6):
```
Comment (back-to) [inline tests/__init__.py]: > Oct 29 2013: Python 2.6.9 final released (security-only)

you don't need to support **python 2.6**

https://github.com/streamlink/streamlink/pull/544
https://www.python.org/dev/peps/pep-0361/#release-schedule

Comment (beardypig) [reply tests/__init__.py]: True, it was still working until recently - but not now, so I'm not going to fix it. #544 was less about removing support for 2.6, but rather not going out of our way to maintain it.

**Comment by beardypig** [comment]
Rebased for #1707

**Comment by back-to** [comment]
since this changes the **default locations,**

it's maybe easier to rebase/merge this,
so new plugins won't be added to the old locations
as it would break this pr all the time.

further changes could be done in a different pr

**Comment by beardypig** [comment]
@back-to yeah, gotta rebase it a lot... didn't even make 1% yet though :D

**Comment by gravyboat** [comment]
Sound good, as soon as the tests pass I'll merge it.

**Comment by gravyboat** [comment]
Restarted the Travis build since it was done and successful but not noting as such here. You'll also have to rebase again due to another PR.

**Comment by gravyboat** [comment]
Thanks @beardypig.

**Pull request merged**